UmbracoFood.Infrastructure/Repositories/RestaurantRepository.cs

```diff
@@ -1,39 +1,53 @@
 ﻿using System.Collections.Generic;
+using System.Data.Common;
+using System.Linq;
+using AutoMapper;
+using Umbraco.Core;
 using UmbracoFood.Core.Interfaces;
 using UmbracoFood.Core.Models;
+using UmbracoFood.Infrastructure.Models.POCO;
 
 namespace UmbracoFood.Infrastructure.Repositories
 {
-    public class RestaurantRepository :IRestaurantRepository
+    public class RestaurantRepository : BaseRepository, IRestaurantRepository
     {
         public void AddRestaurant(Restaurant restaurant)
         {
-            throw new System.NotImplementedException();
+            db.Insert("Restaurants", "Id", Mapper.Map<Restaurant, RestaurantPoco>(restaurant));
         }
 
         public void EditRestaurant(Restaurant restaurant)
         {
-            throw new System.NotImplementedException();
+            var updatedRestaurant = db.SingleOrDefault<RestaurantPoco>("SELECT * FROM Restaurants WHERE Id = @0", restaurant.ID);
+            if (updatedRestaurant == null)
+            {
+                throw new KeyNotFoundException("Restaurant has not been found.");
+            }
+
+            db.Update("Restaurants", "Id", Mapper.Map<Restaurant, RestaurantPoco>(restaurant, updatedRestaurant));
         }
 
         public void RemoveRestaurant(int id)
         {
-            throw new System.NotImplementedException();
+            db.Execute("DELETE FROM Restaurants WHERE Id = @0", id);
         }
 
         public Restaurant GetRestaurant(int id)
         {
-            throw new System.NotImplementedException();
+            var restaurant = db.SingleOrDefault<RestaurantPoco>("SELECT * FROM Restaurants WHERE Id = @0", id);
+            return Mapper.Map<Restaurant>(restaurant);
         }
 
         public IEnumerable<Restaurant> GetActiveRestaurants()
```

Comment:
Z tego co widzę, to
Umbraco.Core.Persistence.Database
nie udostępnia metod, które zwracają IQueryable<T>